tests/tests_suite_test.go

```diff
@@ -381,20 +304,101 @@ func cmdWithRetry(cmd Cmd, expectedResult string, timeout int) bool {
 		}
 		time.Sleep(1 * time.Second)
 	}
-	fmt.Printf("FAIL: '%s' does not match expected result of '%s'\n", result, expectedResult)
+	fmt.Fprintf(GinkgoWriter, "FAIL: '%s' does not match expected result of '%s'\n", result, expectedResult)
 	return false
 }
 
 // gitInit simply invokes 'git init' and verifies the command is successful
 func gitInit() {
-	cmd, err := start("git init")
+	cmd, err := start("git init", "")
 	Expect(err).NotTo(HaveOccurred())
 	Eventually(cmd).Should(Say("Initialized empty Git repository"))
 }
 
 // gitClean destroys the .git directory and verifies the command is successful
 func gitClean() {
-	cmd, err := start("rm -rf .git")
+	cmd, err := start("rm -rf .git", "")
 	Expect(err).NotTo(HaveOccurred())
 	Eventually(cmd).Should(Exit(0))
 }
+
+func registerHTTP(url string, username string, password string, email string) string {
+	registerURL := fmt.Sprintf("%s/v2/auth/register/", url)
+	registerJSONTemplate := "{\"username\": \"%s\",\"password\": \"%s\",\"email\": \"%s\"}"
+	var registerJSON = []byte(fmt.Sprintf(registerJSONTemplate, username, password, email))
+	req, err := http.NewRequest("POST", registerURL, bytes.NewBuffer(registerJSON))
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	registerBody, _ := ioutil.ReadAll(resp.Body)
+	return string(registerBody)
+}
+
+func loginHTTP(url string, username string, password string) string {
+	loginURL := fmt.Sprintf("%s/v2/auth/login/", url)
+	loginJSONTemplate := "{\"username\": \"%s\",\"password\": \"%s\"}"
+	var loginJSON = []byte(fmt.Sprintf(loginJSONTemplate, username, password))
+	req, err := http.NewRequest("POST", loginURL, bytes.NewBuffer(loginJSON))
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	loginBody, _ := ioutil.ReadAll(resp.Body)
+	token := strings.TrimSuffix(strings.Split(string(loginBody), ":")[1], "}")
```

Comment:
A comment or two explaining this block of logic would be helpful.  (Looks like we are creating the deis client profile (json file), putting it in the proper place and making it executable?)  It might even be good to separate this logic out from `loginHTTP` (which would only be responsible for making the http req/getting resp) into another method, say, `createClientProfile`, which will do the steps below.